gallery/gallery.py

```diff
@@ -0,0 +1,205 @@
+import json
+import subprocess
+import sys
+import tarfile
+import zipfile
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from typing import List, NamedTuple, Optional, Union, cast
+from urllib.request import urlopen, urlretrieve
+
+PYPI_INSTANCE = "https://pypi.org/pypi"
+INTERNAL_BLACK_REPO = "__black"  # This shouldn't be conflict with any real package
+
+ArchiveKind = Union[tarfile.TarFile, zipfile.ZipFile]
+
+
+class BlackVersion(NamedTuple):
+    version: str
+    config: Optional[str] = None
+
+
+def get_pypi_download_url(package: str, version: Optional[str]) -> str:
```

Comment:
This is clever. Any idea if this is better than `pip download --no-binary --no-deps`?

Reply:
I was thinking about extending this with options like `--format-pypi-top n`, and for doing that programatically I implemented this in a form of function.